webrtc/RTCPeerConnection-set-description.html

```diff
@@ -0,0 +1,395 @@
+    } else if(countVideoLine(sdp1) !== countVideoLine(sdp2)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function assert_defined(object) {
+    if(object === undefined || object === null)
+      assert_unreached('object must not be undefined or null');
+  }
+
+  function isSimilarSdp(sdp1, sdp2) {
+    assert_defined(sdp1);
+    assert_defined(sdp2);
+
+    if(sdp1.type !== sdp2.type) {
+      return false;
+    } else {
+      // We can't do string comparison to the SDP content,
+      // because RTCPeerConnection may return SDP that is
+      // slightly modified from what is given to it due to
+      // ice candidate events or serialization.
+      // To hack around this, if the SDP strings are not
+      // equal, we simply count the media description lines
+      // and if they are the same, we assume it is the same.
+      return similarMediaDescriptions(sdp1.sdp, sdp2.sdp);
+    }
+  }
+
+  function createOfferAndSetLocal(pc, options) {
+    return pc.createOffer(options)
+    .then(offer => {
+      return pc.setLocalDescription(offer)
+      .then(() => offer)
+    });
+  }
+
+  function createOfferAndSetRemote(pc1, pc2, options) {
+    return createOfferAndSetLocal(pc1, options)
+    .then(offer =>
+      pc2.setRemoteDescription(offer)
+      .then(() => offer));
+  }
+
+  function createOfferAndGetAnswer(pc1, pc2, options) {
+    return createOfferAndSetRemote(pc1, pc2, options)
+    .then(offer =>
+      pc2.createAnswer(offer)
+      .then(answer => [offer, answer]))
+  }
+
+  function createAnswerAndSetLocal(pc1, pc2, options) {
+    return createOfferAndGetAnswer(pc1, pc2, options)
+    .then(([offer, answer]) =>
+      pc2.setLocalDescription(answer)
+      .then(() => [offer, answer]));
+  }
+
+  function createAnswerAndSetRemote(pc1, pc2, options) {
+    return createAnswerAndSetLocal(pc1, pc2, options)
+    .then(([offer, answer]) =>
+      pc1.setRemoteDescription(answer)
+      .then(() => [offer, answer]));
+  }
+
+  test(t => {
+    const pc = new RTCPeerConnection;
+    assert_equals(pc.signalingState, 'stable');
+    assert_equals(pc.localDescription, null);
+    assert_equals(pc.currentLocalDescription, null);
+    assert_equals(pc.pendingLocalDescription, null);
+    assert_equals(pc.remoteDescription, null);
+    assert_equals(pc.currentRemoteDescription, null);
+    assert_equals(pc.pendingRemoteDescription, null);
+  }, 'Initial peer connection state');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection;
+    return createOfferAndSetLocal(pc)
+    .then(offer => {
+      assert_equals(pc.signalingState, 'have-local-offer');
+      assert_true(isSimilarSdp(pc.localDescription, offer));
+      assert_true(isSimilarSdp(pc.pendingLocalDescription, offer));
+      assert_equals(pc.currentLocalDescription, null);
+    });
+  }, 'Create and set offer');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection;
+    return pc.createOffer()
+    .then(offer =>
```

Comment:
What's the purpose of the `createOffer()` call in this test? Is it testing the "Let lastOffer be the result returned by the last call to createOffer." step of the spec? If so, can you adjust the terminology to match the spec, which doesn't mention "default offer"? If the spec has no (useful) terminology, then comments pointing out what's being tested will be necessary.

Note that the spec doesn't define what *lastOffer* is if there was no previous call to `createOffer`, can you file a spec issue?